src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1534,35 +1534,40 @@ private void EmitDefaultValueTypeConstructorCallExpression(BoundCall call)
             FreeOptTemp(tempOpt);
         }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private void EmitStaticCallExpression(BoundCall call, UseKind useKind)
+        private void EmitStaticCall(MethodSymbol method, BoundExpression receiverOpt, ImmutableArray<BoundExpression> arguments, UseKind useKind, SyntaxNode syntax, ImmutableArray<RefKind> refKindsOpt)
         {
-            var method = call.Method;
-            var receiver = call.ReceiverOpt;
-            var arguments = call.Arguments;
-
             Debug.Assert(method.IsStatic);
 
-            EmitArguments(arguments, method.Parameters, call.ArgumentRefKindsOpt);
+            EmitArguments(arguments, method.Parameters, refKindsOpt);
             int stackBehavior = GetCallStackBehavior(method, arguments);
 
             if (method.IsAbstract || method.IsVirtual)
             {
-                if (receiver is not BoundTypeExpression { Type: { TypeKind: TypeKind.TypeParameter } })
+                if (receiverOpt is not BoundTypeExpression { Type.TypeKind: TypeKind.TypeParameter })
                 {
                     throw ExceptionUtilities.Unreachable;
                 }
 
                 _builder.EmitOpCode(ILOpCode.Constrained);
-                EmitSymbolToken(receiver.Type, receiver.Syntax);
+                EmitSymbolToken(receiverOpt.Type, receiverOpt.Syntax);
             }
 
             _builder.EmitOpCode(ILOpCode.Call, stackBehavior);
 
-            EmitSymbolToken(method, call.Syntax,
-                            method.IsVararg ? (BoundArgListOperator)arguments[arguments.Length - 1] : null);
+            EmitSymbolToken(method, syntax,
+                            method.IsVararg ? (BoundArgListOperator)arguments[^1] : null);
 
-            EmitCallCleanup(call.Syntax, useKind, method);
+            EmitCallCleanup(syntax, useKind, method);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void EmitStaticCallExpression(BoundCall call, UseKind useKind)
+        {
+            var method = call.Method;
+            var receiver = call.ReceiverOpt;
+            var arguments = call.Arguments;
+
+            EmitStaticCall(method, receiver, arguments, useKind, call.Syntax, call.ArgumentRefKindsOpt);
```

Comment:
nit: Some of the arguments are inlined and some are not. Consider inlining them all since they are trivial.